corehq/apps/settings/static/settings/js/user_api_keys.js

```diff
@@ -0,0 +1,59 @@
+hqDefine("settings/js/user_api_keys", [
+    "jquery",
+    "knockout",
+    'underscore',
+    "hqwebapp/js/initial_page_data",
+    "hqwebapp/js/crud_paginated_list",
+], function (
+    $,
+    ko,
+    _,
+    initialPageData,
+    CRUDPaginatedList
+) {
+
+    var updateApiKey = function (action, apiKey) {
+        $.ajax({
+            url: "",
+            type: "POST",
+            dataType: 'json',
+            data: {
+                action: action,
+                id: apiKey.itemId,
+            },
+            success: function (data) {
+                apiKey.itemData(data.itemData);
+            },
+        });
+    };
+
+    var ApiKeyModel = function (itemSpec, initRow) {
```

Comment:
Do we really need this model? can we not just add a click binding on the button which calls a function? Is the issue that we can't get the `apiKey` object that way?

Reply:
Yeah that's right - I could of course modify the DOM with jquery or something, but to operate in the "each row is a knockout model" paradigm I think this needs to be a method of that model.

Reply:
Ended up doing that latter approach to get access to some helper methods from the parent model.  It also gets rid of the jquery stuff in this older version, so it feels more uniformly "knockout.js" to me.